draft/src/composite_drafter.rs

```diff
@@ -0,0 +1,109 @@
+use crate::api::*;
+use std::collections::HashMap;
+pub struct CompositeDrafter {
+
+}
+
+// Draft brains will make determinations on how to draft based
+// on various metrics provided to them within card data.
+
+/* A rudimentary brain that counts mana symbols in mana costs and attempts to find cards
+ with average color similar to that of the existing cards in our deck. */
+pub struct DraftColorBrain {
+}
+
+impl DraftColorBrain {
+    // Function that builds a hash map of colors and their weights; higher weights are closer to our
+    // current colors.
+    // TODO: In the python conversion, this state should be kept and updated incrementally, rather
+    // than recalculated wholesale every pick.
+    fn calc_preferences(&self, cards_owned: &[Card]) -> HashMap<BasicMana, i32> {
+        let mut mana_prefs: HashMap<BasicMana, i32> = HashMap::new();
+        for i in 0..cards_owned.len() {
```

Comment:
For each loops (e.g. `for card in cards_owned.iter()`) are nicer, if we don't need to use the index for anything else. (Same for similar loops elsewhere)

Reply:
Not sure why I forgot rust had for each loops.